### PR TITLE
Move cats-effect-3 to main

### PR DESCRIPTION
**Pull request by rossabaker**
The port isn't quite complete, but it's close, and is the final piece of #4106.  The old main is now forked to series/0.22, freeing up this default branch for CE3 development on the 1.0 line.

### CODE REVIEW

**Comment by rossabaker** [comment]
@cquiroz, please see changes around `LawAdapter` and the `EntityCodec` tests.  I had to boost some constraints for lack of `.toIO`, but I think I expressed it properly.

**Comment by rossabaker** [comment]
These fatal warnings on CI only are a nuisance. :rage:

**Comment by rossabaker** [comment]
It appears that I have again neglected to run the tests before flitting off to another branch.

**Comment by rossabaker** [comment]
Ugh, that BatchingExecutor problem is Scala 2.12 only. :fire:

File: testing/src/test/scala/org/http4s/Http4sSuite.scala [hunk 1]
```diff
@@ -25,6 +25,9 @@ import munit._
 /** Common stack for http4s' munit based tests
   */
 trait Http4sSuite extends CatsEffectSuite with DisciplineSuite with munit.ScalaCheckEffectSuite {
+  // The default munit EC causes an IllegalArgumentException in
+  // BatchExecutor on Scala 2.12.
+  override val munitExecutionContext = Http4sSpec.TestExecutionContext
```
Comment (cquiroz) [inline testing/src/test/scala/org/http4s/Http4sSuite.scala:30]: This is interesting. I think I had to use `TestExecutionContext` in a few places too

File: laws/src/main/scala/org/http4s/laws/discipline/LawAdapter.scala [hunk 2]
```diff
@@ -34,11 +34,11 @@ trait LawAdapter {
     propLabel -> PropF.boolean(prop)
 
   def isEqPropF[F[_], A: Arbitrary: Shrink, B: Eq](propLabel: String, prop: A => IsEq[F[B]])(
-      implicit F: Sync[F]): (String, PropF[F]) =
+      implicit F: MonadThrow[F]): (String, PropF[F]) =
```
Comment (cquiroz) [inline laws/src/main/scala/org/http4s/laws/discipline/LawAdapter.scala:37]: This looks right

File: testing/src/test/scala/org/http4s/Http4sSuite.scala (hunk 1 above)
Comment (rossabaker) [reply testing/src/test/scala/org/http4s/Http4sSuite.scala:30]: I discussed with @RaasAhsan a bit last night, and I think we can fork tests and get rid of some of our specialized test thread pools.  But I'm just trying to land this plane.

**Pull request merged**